packages/ui-a11y-utils/src/KeyboardFocusRegion.ts

```diff
@@ -268,62 +230,55 @@ class KeyboardFocusRegion {
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  handleFirstTabbableKeyDown = (event) => {
+  handleFirstTabbableKeyDown = (event: KeyboardEvent) => {
     if (event.keyCode === keycode.codes.tab && event.shiftKey) {
-      // @ts-expect-error ts-migrate(2339) FIXME: Property '_options' does not exist on type 'Keyboa... Remove this comment to see the full error message
-      this._options.onBlur(event)
+      this._options.onBlur?.(event)
     }
   }
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'event' implicitly has an 'any' type.
-  handleLastTabbableKeyDown = (event) => {
+  handleLastTabbableKeyDown = (event: KeyboardEvent) => {
     if (event.keyCode === keycode.codes.tab && !event.shiftKey) {
-      // @ts-expect-error ts-migrate(2339) FIXME: Property '_options' does not exist on type 'Keyboa... Remove this comment to see the full error message
-      this._options.onBlur(event)
+      this._options.onBlur?.(event)
     }
   }
 
   activate() {
     const { defaultFocusElement, shouldContainFocus } = this
-
     if (!this._active) {
       if (defaultFocusElement || shouldContainFocus) {
         if (shouldContainFocus) {
           this._listeners.push(
-            // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ remove(): void; }' is not assi... Remove this comment to see the full error message
-            addEventListener(this.doc, 'keydown', this.handleKeyDown)
+            addEventListener(
+              this.doc,
+              'keydown',
+              this.handleKeyDown as EventListener
+            )
           )
         } else {
           this._listeners.push(
-            // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ remove(): void; }' is not assi... Remove this comment to see the full error message
             addEventListener(
               this.firstTabbable || defaultFocusElement,
               'keydown',
-              this.handleFirstTabbableKeyDown
+              this.handleFirstTabbableKeyDown as EventListener
             )
           )
           this._listeners.push(
-            // @ts-expect-error ts-migrate(2345) FIXME: Argument of type '{ remove(): void; }' is not assi... Remove this comment to see the full error message
             addEventListener(
-              this.lastTabbable || defaultFocusElement,
+              this.lastTabbable || defaultFocusElement!,
```

Comment:
This made me think, that the `findFocusable` util might not be typed correctly, since it might give back and empty array, might it not? Seems logical, that there are no focusable/tabbable elements on a page. If so, then `this.tabbable` might be empty and both `firstTabbable` and `lastTabbable` can be undefined. (might not be the scope of this ticket)

Reply:
yes, you're right, I've added that they can be `undefined`

Reply:
Shall we make a ticket for fixing "findFocusable"?